src/monitor/EventProcessor.ts

```diff
@@ -0,0 +1,114 @@
+import { IDatabase } from '@/database';
+import {
+  ProcessingResult,
+  StakeDepositedEvent,
+  StakeWithdrawnEvent,
+  DelegateeAlteredEvent,
+} from './types';
+import { Logger } from './logging';
+
+export class EventProcessor {
+  constructor(
+    private readonly db: IDatabase,
+    private readonly logger: Logger,
+  ) {}
+
+  async processStakeDeposited(
+    event: StakeDepositedEvent,
+  ): Promise<ProcessingResult> {
+    try {
+      await this.db.createDeposit({
```

Comment:
`StakeDeposited` events are created when a user:

1. First creates a new deposit
2. Adds tokens to an existing deposit

In case 2, I think we'd want to do an update rather than a create, right? So we'd have to check if the depositId already exists in the database, then update the amount.